ublox_cli/src/main.rs

```diff
@@ -1,196 +1,20 @@
-use chrono::prelude::*;
-use clap::{App, Arg};
-use std::convert::TryInto;
-use std::time::Duration;
-use ublox::*;
-
-struct Device {
-    port: Box<dyn serialport::SerialPort>,
-    parser: Parser<Vec<u8>>,
-}
-
-impl Device {
-    pub fn new(port: Box<dyn serialport::SerialPort>) -> Device {
-        let parser = Parser::default();
-        Device { port, parser }
-    }
-
-    pub fn write_all(&mut self, data: &[u8]) -> std::io::Result<()> {
-        self.port.write_all(data)
-    }
-
-    pub fn update<T: FnMut(PacketRef)>(&mut self, mut cb: T) -> std::io::Result<()> {
-        loop {
-            let mut local_buf = [0; 100];
-            let nbytes = self.read_port(&mut local_buf)?;
-            if nbytes == 0 {
-                break;
-            }
-
-            // parser.consume adds the buffer to its internal buffer, and
-            // returns an iterator-like object we can use to process the packets
-            let mut it = self.parser.consume(&local_buf[..nbytes]);
-            loop {
-                match it.next() {
-                    Some(Ok(packet)) => {
-                        cb(packet);
-                    }
-                    Some(Err(_)) => {
-                        // Received a malformed packet, ignore it
-                    }
-                    None => {
-                        // We've eaten all the packets we have
-                        break;
-                    }
-                }
-            }
-        }
-        Ok(())
-    }
-
-    pub fn wait_for_ack<T: UbxPacketMeta>(&mut self) -> std::io::Result<()> {
-        let mut found_packet = false;
-        while !found_packet {
-            self.update(|packet| {
-                if let PacketRef::AckAck(ack) = packet {
-                    if ack.class() == T::CLASS && ack.msg_id() == T::ID {
-                        found_packet = true;
-                    }
-                }
-            })?;
-        }
-        Ok(())
-    }
+use std::io;
```

Comment:
Would it make sense to move this to a different example crate, or add it as another option alongside the previous code?

Reply:
Yeah was just testing for our use case, this has been reverted though. Passes CI as well with reverted MSRV